Python/Product/Analysis/LanguageServer/Structures.cs

```diff
@@ -320,23 +324,33 @@ public struct ExecuteCommandCapabilities { public bool dynamicRegistration; }
     }
 
     [Serializable]
-    public class TextDocumentClientCapabilities {
+    public struct TextDocumentClientCapabilities {
```

Comment:
I believe this will prevent us from changing the contents of this struct/class in the future. I'd rather it stay as a class (and probably should be sealed).

Reply:
(Similar for the other types in this file)

Reply:
I agree that we need to declare clear rules on how types for serialization are declared, but I'm not sure how classes would help with versioning. To be detached from protocol changes, we need to remove all references to protocol data types from public part of API and/or inject a group of abstractions.

The benefits of structs are that they are not nullable, which saves us from endless null checks, enforces default values and makes `null` a real expected value instead of a sign that we haven't serialized something.

The benefit of classes is that they are safely mutable, but I'm against mutable protocol data types. Another benefit is that class loses its default constructor when non-default is presented, which will immediately give us deseriazliation error if default value isn't properly defined.

@MikhailArkhipov , what is your opinion?